### PR TITLE
User analysis

### PR DESCRIPTION
**Pull request by SteRiccio**

### CODE REVIEW

File: server/modules/analysis/repository/processingChainRepository.js [hunk 1]
```diff
@@ -119,3 +119,12 @@ export const deleteChainsWithoutCycles = async (surveyId, client = db) =>
     DELETE FROM ${getSurveyDBSchema(surveyId)}.processing_chain
     WHERE jsonb_array_length(props->'${ProcessingChain.keysProps.cycles}') = 0
   `)
+
+// ===== GRANT PRIVILEGES
+export const grantUpdateOnProcessingChainStatusToUser = async (surveyId, userName, client = db) =>
```
Comment (minotogna) [inline server/modules/analysis/repository/processingChainRepository.js]: should we shorten the name? e.g. grantUpdateToAnalysisUser?

File: server/modules/surveyRdb/repository/nodeAnalysisTableRepository.js [hunk 2]
```diff
@@ -20,3 +21,11 @@ export const createNodeAnalysisTable = async (surveyId, client) => {
     )
   `)
 }
+
+// ===== GRANT PRIVILEGES
+export const grantWriteOnNodeAnalysisToUser = async (surveyId, userName, client) =>
```
Comment (minotogna) [inline server/modules/surveyRdb/repository/nodeAnalysisTableRepository.js]: should we call this name grantWriteToAnalysisUser?

File: server/modules/surveyRdb/repository/schemaRdbRepository.js [hunk 3]
```diff
@@ -6,3 +6,13 @@ export const dropSchema = async (surveyId, client = db) =>
 
 export const createSchema = async (surveyId, client = db) =>
   await client.query(`CREATE SCHEMA ${SchemaRdb.getName(surveyId)}`)
+
+export const grantSchemaSelectToUser = async (surveyId, userName, client = db) => {
```
Comment (minotogna) [inline server/modules/surveyRdb/repository/schemaRdbRepository.js]: and this grantSelectToAnalysisUser?

File: server/modules/user/manager/userAnalysisManager.js [hunk 4]
```diff
@@ -1 +1,24 @@
-export { insertOrUpdateUserAnalysis } from '../repository/userAnalysisRepository'
+import { db } from '@server/db/db'
```
Comment (minotogna) [inline server/modules/user/manager/userAnalysisManager.js]: should this manager to to analysis module?

File: server/modules/user/repository/userAnalysisRepository.js [hunk 5]
```diff
@@ -1,15 +1,33 @@
 import { db } from '@server/db/db'
```
Comment (minotogna) [inline server/modules/user/repository/userAnalysisRepository.js]: and this repository to analysis module as well?

File: server/db/migration/public/migrations/sqls/20180709092931-create-user-table-up.sql [hunk 6]
```diff
@@ -41,4 +41,14 @@ CREATE TABLE
...
+
+CREATE TABLE
+  user_analysis
+(
+  survey_id     BIGINT      NOT NULL REFERENCES survey("id") ON DELETE CASCADE,
+  name          VARCHAR(47) NOT NULL DEFAULT 'arena_data_' || uuid_generate_v4(),
+  password      VARCHAR(36) NOT NULL DEFAULT uuid_generate_v4(),
+
+  PRIMARY KEY (survey_id)
+);
```
Comment (minotogna) [inline server/db/migration/public/migrations/sqls/20180709092931-create-user-table-up.sql]: and this to analysis migration ?

File: core/user/userAnalysis.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import * as R from 'ramda'
```
[the remark this answers is not in the document]
Comment (SteRiccio) [reply core/user/userAnalysis.js]: right, I'll move it

File: server/modules/analysis/repository/processingChainRepository.js (hunk 1 above)
Comment (SteRiccio) [reply server/modules/analysis/repository/processingChainRepository.js]: ok

File: server/modules/surveyRdb/repository/nodeAnalysisTableRepository.js (hunk 2 above)
Comment (SteRiccio) [reply server/modules/surveyRdb/repository/nodeAnalysisTableRepository.js]: ok

File: server/modules/user/manager/userAnalysisManager.js (hunk 4 above)
Comment (SteRiccio) [reply server/modules/user/manager/userAnalysisManager.js]: ok

File: server/modules/user/repository/userAnalysisRepository.js (hunk 5 above)
Comment (SteRiccio) [reply server/modules/user/repository/userAnalysisRepository.js]: yes, of course

File: server/db/migration/public/migrations/sqls/20180709092931-create-user-table-up.sql (hunk 6 above)
Comment (SteRiccio) [reply server/db/migration/public/migrations/sqls/20180709092931-create-user-table-up.sql]: right

**Pull request merged**